meilisearch-http/src/snapshot.rs

```diff
@@ -0,0 +1,118 @@
+use std::path::{PathBuf, Path};
+use std::fs::{create_dir_all, File};
+use tar::{Builder, Archive};
+use flate2::write::GzEncoder;
+use flate2::read::GzDecoder;
+use flate2::Compression;
+use std::thread;
+use std::time::{Duration};
+use tempfile::TempDir;
+use log::{error, warn};
+use crate::error::Error;
+use crate::Data;
+
+fn pack(src: &Path, dest: &Path) -> Result<(), Box<dyn std::error::Error>> {
+    let f = File::create(dest)?;
+    let mut gz_encoder = GzEncoder::new(f, Compression::default());
+    {
+        let mut tar_encoder = Builder::new(&mut gz_encoder);
+
+        tar_encoder.append_dir_all(".", src)?;
+        tar_encoder.finish()?;
+    }
+    gz_encoder.finish()?;
+
+    Ok(())
+}
+
+fn unpack(src: &Path, dest: &Path) -> Result<(), Error> {
+    let f = File::open(src)?;
+    let gz = GzDecoder::new(f);
+    let mut ar = Archive::new(gz);
+
+    create_dir_all(dest)?;
+    ar.unpack(dest)?;
+
+    Ok(())
+}
+
+pub fn import_snapshot(db_path: &str, snapshot_path: &Path) -> Result<(), Error> {
+    let db_path = Path::new(db_path);
+
+    if db_path.exists() {
+        warn!("database already exist at {:?}, skiping snapshot importation", db_path);
```

Comment:
here I puted a warning if a database already exist, it's not a fatal error to let user use this command in his deployment strategy (drop and restart...). It could be a log info, I don't know which is better. 😄

Reply:
what db should you take then? I think this should abort execution